File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int result = 0;
        
        int left = 0;
        int right = 0;
        
        while (right < prices.length) {
            result = Math.max(result, prices[right] - prices[left]);
            
            if (prices[right] < prices[left]) {
                left = right;
            }
            
            right++;
        }
        return result;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    low := 0
    maxProf := 0
    
    for i, price := range(prices) {
        if (i == 0) {
            low = price
            continue
        }
        
        if (low > price) {
            low = price
        }
        
        maxProf = max(maxProf, price - low)
    }
    
    return maxProf
}